### PR TITLE
CLDR-14597 Restrict Organizations to specified locales

### PR DESCRIPTION
**Pull request by btangmu**
-New functions u.hasLocalePermission, userHasLocale, orgHasLocale
-New class LocaleNormalizer with warnings reportable as html or plain text
-New unit test TestLocaleNormalizer
-Move code from SurveyMain.doGet into new subroutines createNewAndLogin, reportNormalizationWarning
-Remove obsolete UserRegistry.EXPERT, Level.expert
-Replace userIsExpert with userIsManagerOrStronger
-Revise unit tests that used expert users
-Make private the version of getGroup that is only used privately
-Simplify UserList.setLocales and UserRegistry.setLocales, use CookieSession instead of WebContext
-Fix unrelated js warning when json.status not defined
-Catch transient exception in setValueFromResolver
-Remove dead code
-Fix deprecation and other warnings
-Comments

CLDR-14597

- [ ] This PR completes the ticket.

### CODE REVIEW

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserRegistry.java [hunk 1]
```diff
@@ -186,8 +184,10 @@ private synchronized void notify(User u) {
 
     /**
      * List of all user levels - for UI presentation
-     **/
-    public static final int ALL_LEVELS[] = { ADMIN, TC, MANAGER, EXPERT, VETTER, STREET, LOCKED };
+     *
+     * Code related to UserRegistry.EXPERT removed 2021-05-18 per CLDR-14597
+     */
+    public static final int ALL_LEVELS[] = { ADMIN, TC, MANAGER, VETTER, STREET, LOCKED };
```
Comment (srl295) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserRegistry.java:191]: probably a separate PR (or even ticket), but this should ideally be generated from VoteResolver.Level

Comment (srl295) [reply tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserRegistry.java:191]: i'll file a separate issue

**Review by srl295 — DISMISSED** [review]
LG so far

**Comment by btangmu** [comment]
I see there's a test error in TestVoteResolver -- I'll work on that

**Comment by btangmu** [comment]
@srl295 FYI if the tests pass, I'll squash this and make it ready for review

**Comment by btangmu** [comment]
I just realized there's one unfinished part of the ticket. When a Manager sets the locales for a Vetter, there should be a warning if any of the specified locales (other than "*") isn't covered by the Organization.

These changes already ensure that a Vetter can't modify locales not covered by their org, but there's no warning to the Manager yet.

I'd like to go ahead with this PR anyway and then add the warning to the Manager.

**Review by srl295 — APPROVED** [review]
LGTM!

**Comment by btangmu** [comment]
This PR has been stuck on the "Running JS Tests" check for 2 hours and 46 minutes! I'll try restarting

**Pull request merged**